lib/jekyll/static_file.rb

```diff
@@ -104,7 +100,7 @@ def to_liquid
     end
 
     def data
-      @data ||= {}
+      @data ||= @site.frontmatter_defaults.all(relative_path, type)
```

Comment:
This isn't guaranteed to run if `@data` is set by some other process. Should we just do this in `#initialize`?